### PR TITLE
add SPP support for G-F deep convection

### PR DESCRIPTION
**Pull request by JiliDong-NOAA**
This PR is co-authored with Georg Grell and @haiqinli, including:
1. add stochastic parameterization perturbations support to Grell-Freitas deep convection parameterization by passing stochastic patterns to mass transport, vertical mass flux profile and closures
2. fix reproducibility issue in G-F deep convection at high resolutions (ichoice option switched to 0)
Related PRs for FV3 and ufs-weather-model will be submitted shortly.

### CODE REVIEW

**Review by haiqinli — APPROVED** [review]

**Review by Qingfu-Liu — APPROVED** [review]

File: physics/cu_gf_driver.F90 [hunk 1]
```diff
@@ -313,9 +318,18 @@ subroutine cu_gf_driver_run(ntracer,garea,im,km,dt,flag_init,flag_restart,&
...
-     rand_mom(:)    = 0.
-     rand_vmas(:)   = 0.
-     rand_clos(:,:) = 0.
+     if (spp_cu_deep == 0) then
+       rand_mom(:)    = 0.
+       rand_vmas(:)   = 0.
+       rand_clos(:,:) = 0.
+     else 
+       do i=1,im
+         spp_wts_cu_deep_tmp=min(max(-1.0, spp_wts_cu_deep(i,1)),1.0)
```
Comment (dustinswales) [inline physics/cu_gf_driver.F90]: It's best to be explicit with the types you are passing to intrinsics.
Replace min(max(-1.0, spp_wts_cu_deep(i,1)),1.0)
With min(max(-1.kind_phys, spp_wts_cu_deep(i,1)),1._kind_phys)

**Review by dustinswales — CHANGES_REQUESTED** [review]
Looks good. Just a small request.

File: physics/cu_gf_driver.F90 (hunk 1 above)
Comment (JiliDong-NOAA) [reply physics/cu_gf_driver.F90]: updated as requested. Thanks for the suggestion @dustinswales

**Review by dustinswales — APPROVED** [review]
@JiliDong-NOAA Thanks for making this change.

**Review by grantfirl — APPROVED** [review]

**Comment by zach1221** [comment]
@grantfirl testing is complete on ufs-wm PR #1876. Are you able to merge this PR?

**Pull request merged**